resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -61,10 +61,8 @@ private[spark] class KubernetesClusterSchedulerBackend(
   private val runningExecutorsToPods = new mutable.HashMap[String, Pod]
   // Indexed by executor pod names and guarded by RUNNING_EXECUTOR_PODS_LOCK.
   private val runningPodsToExecutors = new mutable.HashMap[String, String]
-  // TODO(varun): Get rid of this lock object by my making the underlying map a concurrent hash map.
-  private val EXECUTOR_PODS_BY_IPS_LOCK = new Object
-  // Indexed by executor IP addrs and guarded by EXECUTOR_PODS_BY_IPS_LOCK
-  private val executorPodsByIPs = new mutable.HashMap[String, Pod]
+  private val executorPodsByIPs: concurrent.Map[String, Pod] = new
```

Comment:
@satybald Thanks for bringing this up. Please note that SI-7943 doesn't really affect the thread correctness of our code. It is specific to only the TrieMap implementation. We are already using the java.util.concurrent.ConcurrentHashMap underneath and using the scala.concurrent.Map trait on top for access.

We should not use the trait nevertheless per the community recommendation to avoid any future maintenance or merging of it upstream.